[go_router] rewrite readme and examples (#2382)
flutter/packages: packages/go_router/README.md

> # A Declarative Routing Package for Flutter

Reviewer: nit: most of the other projects in flutter/packages put either the package name or title-cased name of the package in the headline

```suggestion
# go_router

A declarative routing package for Flutter.
```

> # Go_router, A Declarative Routing Package for Flutter

Reviewer: nit: go_router or GoRouter?

```suggestion
# go_router, A Declarative Routing Package for Flutter
```

> This package builds on top of Flutter Router API and provides convenient

Reviewer: ```suggestion
This package builds on top of the Flutter framework's Router API and provides convenient
```

    ],
  );
}
```

Still not sure how to proceed? See [examples](https://github.com/flutter/packages/tree/main/packages/go_router/example)
for complete runnable examples or visit [API documentation](https://pub.dev/documentation/go_router/latest/go_router/go_router-library.html)
for more detailed explanations.

### Migration guides

[Migrating to 2.0](https://flutter.dev/go/go-router-v2-breaking-changes)<br/>
[Migrating to 2.5](https://flutter.dev/go/go-router-v2-5-breaking-changes)<br/>
[Migrating to 3.0](https://flutter.dev/go/go-router-v3-breaking-changes)<br/>
[Migrating to 4.0](https://flutter.dev/go/go-router-v4-breaking-changes)<br/>

> ### [Change Log](https://github.com/flutter/packages/blob/main/packages/go_router/CHANGELOG.md)

Reviewer: ```suggestion
### Changelog
See the [Changelog](https://github.com/flutter/packages/blob/main/packages/go_router/CHANGELOG.md) for a list of new features and breaking changes.
```
